src/arb/Complex.jl

```diff
@@ -1616,135 +1616,127 @@ end
-    end
+function set!(x::ComplexFieldElemOrPtr, y::ComplexFieldElemOrPtr)
+  @ccall libflint.acb_set(x::Ref{ComplexFieldElem}, y::Ref{ComplexFieldElem})::Nothing
+end
 
-    function _acb_set(x::($typeofx), y::BigFloat)
-      r = ccall((:acb_real_ptr, libflint), Ptr{RealFieldElem}, (($passtoc), ), x)
-      _arb_set(r, y)
-      i = ccall((:acb_imag_ptr, libflint), Ptr{RealFieldElem}, (($passtoc), ), x)
-      zero!(i)
-    end
+function set!(x::ComplexFieldElemOrPtr, y::ComplexFieldElemOrPtr, p::Int)
+  @ccall libflint.acb_set_round(x::Ref{ComplexFieldElem}, y::Ref{ComplexFieldElem}, p::Int)::Nothing
+end
 
-    function _acb_set(x::($typeofx), y::BigFloat, p::Int)
-      r = ccall((:acb_real_ptr, libflint), Ptr{RealFieldElem}, (($passtoc), ), x)
-      _arb_set(r, y, p)
-      i = ccall((:acb_imag_ptr, libflint), Ptr{RealFieldElem}, (($passtoc), ), x)
-      zero!(i)
-    end
+function set!(x::ComplexFieldElemOrPtr, y::AbstractString, p::Int)
+  r = _real_ptr(x)
+  set!(r, y, p)
+  i = _imag_ptr(x)
+  zero!(i)
+end
 
-    function _acb_set(x::($typeofx), y::Int, z::Int, p::Int)
-      ccall((:acb_set_si_si, libflint), Nothing,
-            (($passtoc), Int, Int), x, y, z)
-      ccall((:acb_set_round, libflint), Nothing,
-            (($passtoc), ($passtoc), Int), x, x, p)
-    end
+function set!(x::ComplexFieldElemOrPtr, y::BigFloat)
+  r = _real_ptr(x)
+  set!(r, y)
+  i = _imag_ptr(x)
+  zero!(i)
+end
 
-    function _acb_set(x::($typeofx), y::RealFieldElem, z::RealFieldElem)
-      ccall((:acb_set_arb_arb, libflint), Nothing,
-            (($passtoc), Ref{RealFieldElem}, Ref{RealFieldElem}), x, y, z)
-    end
+function set!(x::ComplexFieldElemOrPtr, y::BigFloat, p::Int)
+  r = _real_ptr(x)
+  set!(r, y, p)
+  i = _imag_ptr(x)
+  zero!(i)
+end
 
-    function _acb_set(x::($typeofx), y::RealFieldElem, z::RealFieldElem, p::Int)
-      _acb_set(x, y, z)
-      ccall((:acb_set_round, libflint), Nothing,
-            (($passtoc), ($passtoc), Int), x, x, p)
-    end
+function set!(x::ComplexFieldElemOrPtr, y::Real, p::Int)
+  r = _real_ptr(x)
+  set!(r, y, p)
+  i = _imag_ptr(x)
+  zero!(i)
+end
 
-    function _acb_set(x::($typeofx), y::QQFieldElem, z::QQFieldElem, p::Int)
-      r = ccall((:acb_real_ptr, libflint), Ptr{RealFieldElem}, (($passtoc), ), x)
-      _arb_set(r, y, p)
-      i = ccall((:acb_imag_ptr, libflint), Ptr{RealFieldElem}, (($passtoc), ), x)
-      _arb_set(i, z, p)
-    end
+function set!(x::ComplexFieldElemOrPtr, y::Tuple{Int, Int}, p::Int)
+  @ccall libflint.acb_set_si_si(x::Ref{ComplexFieldElem}, y[1]::Int, y[2]::Int)::Nothing
+end
 
-    function _acb_set(x::($typeofx), y::T, z::T, p::Int) where {T <: AbstractString}
-      r = ccall((:acb_real_ptr, libflint), Ptr{RealFieldElem}, (($passtoc), ), x)
-      _arb_set(r, y, p)
-      i = ccall((:acb_imag_ptr, libflint), Ptr{RealFieldElem}, (($passtoc), ), x)
-      _arb_set(i, z, p)
-    end
+function set!(x::ComplexFieldElemOrPtr, y::Tuple{Int, Int}, p::Int)
```

Comment:
One more change here that I think we should consider to retain for `_acb_set` methods: pairing "real+imag" arguments together into a single 2-tuple argument. That avoids ambiguities with those `p::Int` precision arguments...